### PR TITLE
Validation rule authenticated user do not match provided sender id

### PR DESCRIPTION
**Pull request by Mech0z**
**Synopsis**
As a grid access provider
When I post a CIM/XML where sender_MarketParticipant_mRID is not equal to my own GLN
Then I want my request to be rejected with reason code E0I

**Tech note**
Actor GLN can be extracted from request token using `Energinet.DataHub.Core.App.FunctionApp.Middleware.ActorMiddleware`

**Acceptance criteria**:
* [ ] Requests where actor GLN in request payload _sender_MarketParticipant_mRID_ does not match the actor GLN in the token of the request are rejected
* [ ] Rejection must be sent to the actor of the request identified by the GLN in the token of the request.

Reject message and code:
Reason code: E0I
GLN code <1> not allowed: The specified grid operator is not the owner and therefore not authorized to update or stop the charge <2>. 
<1>=#SenderId#
<2>=#ChargeTypeId#

## References

* #1230

### CODE REVIEW

File: docs/business-processes/validations.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ The following validation rules are currently implemented in the charge domain.
 |VR.902-2*|Update and stop charge link is not yet implemented|D13|N/A|X|
 |VR.903|The Tax indicator for a charge cannot be updated|D14|Tariff|N/A|
 |VR.905|As gaps are not allowed in a Charge timeline, e.g. when a charge is stopped, any attempts to update it with an effective date after the stop date, must be rejected.|D14|All|N/A|
+|VR.907|The provided market participant sender id do not match authenticated user|E0I|All|N/A|
```
Comment (HenrikSommer) [inline docs/business-processes/validations.md:34]:
```suggestion
|VR.907|The provided market participant sender id does not match the authenticated user|E0I|All|N/A|
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs [hunk 2]
```diff
@@ -63,12 +63,11 @@ internal static void ConfigureServices(IServiceCollection serviceCollection)
         {
             serviceCollection.AddScoped(typeof(IClock), _ => SystemClock.Instance);
             serviceCollection.AddLogging();
+            serviceCollection.AddJwtTokenSecurity();
             serviceCollection.AddScoped<CorrelationIdMiddleware>();
             serviceCollection.AddScoped<FunctionTelemetryScopeMiddleware>();
             serviceCollection.AddScoped<MessageMetaDataMiddleware>();
             serviceCollection.AddScoped<FunctionInvocationLoggingMiddleware>();
-            serviceCollection.AddJwtTokenSecurity();
-            //serviceCollection.AddActorContext();
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs:71]: I consider the ActorContext relatively unrelated to the security. Why not just call `AddActorContext()` here?

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs:71]: changed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 3]
```diff
@@ -137,6 +137,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string UpdateChargeMustHaveEffectiveDateBeforeOrOnStopDateErrorText =
             "Charge ID {{DocumentSenderProvidedChargeId}} has been stopped and thus cannot be updated as per {{ChargeStartDateTime}}.";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.AuthenticatedUserMustMatchMarketParticipantSenderIdRule)]
+        public const string AuthenticatedUserMustMatchMarketParticipantSenderIdRuleErrorText =
+            "GLN code {{DocumentSenderId}} not allowed: The specified grid operator is not the owner and therefore not authorized to update or stop the charge {{ChargeType}}.";
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs:142]:
```suggestion
            "GLN code {{DocumentSenderId}} not allowed: The specified grid operator is not the owner and therefore not authorized to create, update, stop or cancel a stop for the charge {{ChargeType}}.";
```

**Review by HenrikSommer — COMMENTED** [review]
Looks good so far. I've left a few comments.

**Comment by Mech0z** [comment]
Closing this as we are implementing sync validation instead

**Pull request closed without merging**